NotesApp/Scenes/Editor/EditorViewController.swift

```diff
@@ -0,0 +1,206 @@
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        return textView
+    }()
+
+    // MARK: - Lifecycle
+    init(viewModel: EditorViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+        setupBindings()
+        viewModel.loadData()
+        saveEmptyNoteIfNeeded()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+
+    // MARK: - UI Setup
+    private func setupUI() {
+        view.backgroundColor = UIColor(hex: "#252525")
+        
+        setupCustomBackButton()
+        
+        titleTextView.addDoneButtonToolbar()
+        bodyTextView.addDoneButtonToolbar(onDone: (target: self, action: #selector(customDoneAction)))
+        
+        view.addSubview(titleTextView)
+        view.addSubview(bodyTextView)
+        
+        titleTextView.translatesAutoresizingMaskIntoConstraints = false
+        bodyTextView.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            titleTextView.topAnchor.constraint(equalTo: customBackButton.bottomAnchor, constant: 20),
+            titleTextView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            titleTextView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            
+            bodyTextView.topAnchor.constraint(equalTo: titleTextView.bottomAnchor, constant: 10),
+            bodyTextView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            bodyTextView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            bodyTextView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 20)
+        ])
+    }
+
+    private func setupCustomBackButton() {
+        customBackButton.addAction(UIAction(handler: { [weak self] _ in
+            self?.backButtonTapped()
+        }), for: .touchUpInside)
+        view.addSubview(customBackButton)
+        
+        customBackButton.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            customBackButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            customBackButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 8),
+            customBackButton.widthAnchor.constraint(equalToConstant: 48),
+            customBackButton.heightAnchor.constraint(equalToConstant: 48),
+        ])
+    }
```

Comment:
it is not correct having everything from action to constraints inside one setup function